tests/Unit/DataProcessorTest.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+namespace WMDE\Fundraising\Frontend\Tests\Unit\Cli;
+
+use Piwik\Plugins\CampaignVisitorsByTime\Test\TestDataProcessor;
+
+/**
+ * @covers \Piwik\Plugins\CampaignVisitorsByTime\DataProcessor
+ */
+class DataProcessorTest extends \PHPUnit\Framework\TestCase {
+
+	const TEST_REFERER = 'test_referer';
+	const TEST_REFERER_ALT = 'test_referer_alt';
+	const TEST_HOUR = 5;
+	const TEST_MINUTES = '00';
+	const TEST_MINUTES_ALT = '15';
+
+	public function testWhenMultipleKeywordsAreUsed_campaignValuesAreCalculatedCorrectly() {
+		$rows = $this->newZendDbRowsMock( $this->getMockDatabaseData() );
+		$dataProcessor = new TestDataProcessor();
+		$dataProcessor->processDatasets( $rows );
+		$this->assertEquals(
+			40,
+			$dataProcessor->getCampaigns()[self::TEST_REFERER][$this->getStoredHour(
+				self::TEST_HOUR,
+				self::TEST_MINUTES
+			)],
+			'Total for given time with timezone offset should sum up all keywords for specific 15 minute block for a specific referrer'
+		);
+	}
+
+	public function testWhenKeywordsAreStored_visitorNumbersAreCorrectlyAssignedToTime() {
+		$rows = $this->newZendDbRowsMock( $this->getMockDatabaseData() );
+		$dataProcessor = new TestDataProcessor();
+		$dataProcessor->processDatasets( $rows );
+		$keywordData = $dataProcessor->getCampaignKeywordData( self::TEST_REFERER );
+		$this->assertEquals(
+			33,
+			$keywordData['testkeyword04'][$this->getStoredHour( self::TEST_HOUR, self::TEST_MINUTES_ALT )],
+			'Individual keywords store their own value per campaign for a specific given 15 minutes block'
+		);
+	}
+
```

Comment:
please add the test `WhenMultipleKeywordsAreUsed_theirTotalsAreAdded`, using a different hour for one of the keywords and checking the `total` key for the whole campaign and the keyword.